airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -1073,32 +1105,68 @@ def __init__(
         pipeline_exec_arn: str,
         wait_for_completion: bool = False,
         check_interval: int = CHECK_INTERVAL_SECOND,
+        waiter_max_attempts: int = 9999,
```

Comment:
For some future change, I wonder if some global constant MAX_ATTEMPTS may be useful since you seem to like using 9999.  Or maybe even a config entry to let the users set a global default for that?

Reply:
We should certainly uniformize this a bit between operators.
Currently, some wait forever by default (like this one), some others have a set value...
It's not an easy task though, because different operators might have different "expected waiting times".
This could be achieved by setting different poll intervals, but always the same max attempts maybe ?
And maybe a global conf to set it yeah ?

Reply:
I think personally I'd do it the other way, have a conf value to set a global interval and adjust the max attempts, but either way, I think it's the right direction to consider in the future.

Approved this as-is, that's a change for another day.